addons/l10n_it_edi_sdicoop/models/ir_attachment.py

```diff
@@ -0,0 +1,26 @@
+# -*- coding:utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import _, models
+from odoo.exceptions import UserError
+
+class IrAttachment(models.Model):
+    _inherit = 'ir.attachment'
+
+    def unlink(self):
+        """ Unlinking of FatturaPA attachments of already sent to EDI moves is forbidden.
+            To verify that, we check the name and whether we have an ID for the EDI transaction,
+            meaning that the SdI already answered us at least once.
+        """
+        errors = []
+        fattura_pa = self.env['account.edi.format']
+        fattura_pa_attachments = self.filtered(
+            lambda attachment: attachment.res_model == 'account.move'
+                               and fattura_pa._check_filename_is_fattura_pa(attachment.name))
```

Comment:
@lordkrandel Why not just check l10n_it_edi_attachment_id ?

Reply:
(btw, we should change the many2one ir.attachment to Binary fields)

Reply:
There is just one l10n_it_edi_attachment_id, while there may be many generated attachments in case of multiple rejection.  

The reason is that I personally don't want any of the FatturaPA attachments to be deleted.
Even if incorrect, the attachment has been sent to the SdI and can be used as a reference or for troubleshooting.
The users try and delete attachments just to see what happens in case of error.

Can change if you want.

Reply:
@lordkrandel The user might be confused by that many attachments to e.g. which one is the valid one and would want to delete some.  I wonder if an ondelete restrict would do the trick as well.  

(P.S. no -> master)

Reply:
So?

Reply:
It still does not make sense to me to forbid the user to delete rejected attachments.

Reply:
@lordkrandel Perhaps if you explain a situation where it was super useful to keep an old attachment, that would help to decide the best option here.

Reply:
You probably didn't have to go through the history of invoices with deleted attachments to understand what happened to a client's invoice like I did. It does not make any sense or gain to delete rejected invoices other than space on the cloud (negligible) or confusion in the client's head.

I'll modify this, because I have an open generous heart, but I still do not agree.

Reply:
When troubleshooting an invoice, you see an only attachment, you won't notice what problems the client had in trying to handle it because the attachment is gone forever.

So you won't be able to see they just solved a global problem with a local solution (like shortening a field that was too long and we didn't trim) or how the XML was actually made when they had a certain error.

We still don't have in place such a clean EDI that prevents errors from happening, so when they happen we need to have the full history.

Reply:
>  I'll modify this, because I have an open generous heart, but I still do not agree.

Done.